File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.js
/**
 * @param {number[]} capacity
 * @param {number[]} rocks
 * @param {number} additionalRocks
 * @return {number}
 */
var maximumBags = function(capacity, rocks, additionalRocks) {
  let rocksNeeded = new Array(capacity.length).fill(0);
  
  for (let i = 0; i < capacity.length; i++) {
    rocksNeeded[i] = capacity[i] - rocks[i];
  }
  
  let maxBags = 0;
  rocksNeeded.sort((a,b) => a-b);
  
  for (let r = 0; r < rocksNeeded.length; r++) {
    if (rocksNeeded[r] === 0) {
      maxBags++;
    } else if (additionalRocks >= rocksNeeded[r]) {
        additionalRocks -= rocksNeeded[r];
        maxBags++;
    }
  }
  
  return maxBags;
};

File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
      rocks_needed = [0]*len(rocks)
      for i in range(len(rocks)):
        rocks_needed[i] = capacity[i]-rocks[i]
      rocks_needed.sort()
      
      max_bags = 0
      
      for r in range(len(rocks_needed)):
        if rocks_needed[r] == 0:
          max_bags += 1
        elif additionalRocks >= rocks_needed[r]:
          additionalRocks -= rocks_needed[r]
          rocks_needed[r] = 0
          max_bags += 1  
      
      return max_bags